content/tutorials/use-tokens/create-an-automated-market-maker.md

```diff
@@ -0,0 +1,265 @@
+    - You can also read along and use the interactive steps in your browser without any setup.
+- You should have a basic understanding of how [tokens](tokens.html) work in the XRP Ledger.
+- You may want to read about [Automated Market Makers in the XRP Ledger](automated-market-makers.html) first.
+
+
+## Example Code
+
+Complete sample code for all of the steps of these tutorials is available under the [MIT license](https://github.com/XRPLF/xrpl-dev-portal/blob/master/LICENSE).
+
+- See [Code Samples: Create an AMM](https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples/create-amm/) in the source repository for this website.
+
+
+## Steps
+{% set n = cycler(* range(1,99)) %}
+{% set use_network = "Devnet" %}
+
+### {{n.next()}}. Connect to the network
+
+You must be connected to the network to query it and submit transactions. The following code shows how to connect to a public {{use_network}} server using a supported [client library](client-libraries.html):
+
+<!-- MULTICODE_BLOCK_START -->
+
+_JavaScript_
+
+{{ include_code("_code-samples/create-amm/js/connect.js", language="js") }}
+
+<!-- MULTICODE_BLOCK_END -->
+
+For this tutorial, click the following button to connect:
+
+{% include '_snippets/interactive-tutorials/connect-step.md' %}
+
+### {{n.next()}}. Get credentials
+
+To transact on the XRP Ledger, you need an address and secret key, and some XRP. For development purposes, you can get these on the [{{use_network}}](parallel-networks.html) using the following interface:
+
+{% include '_snippets/interactive-tutorials/generate-step.md' %}
+
+When you're [building production-ready software](production-readiness.html), you should use an existing account, and manage your keys using a [secure signing configuration](set-up-secure-signing.html). The following code shows how to get a `Wallet` instance using either the faucet or a seed provided by environment variable:
+
+<!-- MULTICODE_BLOCK_START -->
+
+_JavaScript_
+
+{{ include_code("_code-samples/create-amm/js/create-amm.js", language="js", start_with="// Get credentials", end_before="// Acquire tokens") }}
+
+<!-- MULTICODE_BLOCK_END -->
+
+
+### {{n.next()}}. Select and acquire assets
+
+As the creator of an AMM, you are also the first liquidity provider and you have to supply it with a starting pool of assets. Other users of the XRP Ledger can also become liquidity providers by supplying assets after the AMM exists. It's important to choose assets especially carefully because, as a liquidity provider for an AMM, you are supplying some amounts of both for users to swap between. If one of the AMM's assets becomes worthless, other users can use the AMM to trade for the other asset, leaving the AMM (and thus, its liquidity providers including you) holding only the worthless one. Technically, the AMM always holds some positive amount of both assets, but the amounts can be very small.
```

Comment:
```suggestion
As the creator of an AMM, you are also the first liquidity provider and you have to supply it with a starting pool of assets. Other users of the XRP Ledger can also become liquidity providers by supplying assets after the AMM exists. It's crucial to choose assets carefully because, as a liquidity provider for an AMM, you are supplying some amounts of both for users to swap between. If one of the AMM's assets becomes worthless, other users can use the AMM to trade for the other asset, leaving the AMM (and thus, its liquidity providers including you) holding only the worthless one. Technically, the AMM always holds some positive amount of both assets, but the amounts can be very small.
```